### DISCUSSION TITLE
Select until the end of the line

### DISCUSSION THREAD

**Discussion by jgarte** [meow-edit/meow#604] [General]
Hi,

How do I select from the current position until the end of the line.

It looks like `meow-tutor` doesn't include an explicit example on how to do this.

It does include the following but that is to delete until the end of the line and not select until the end of the line 🦆 

>  Note: Pressing s without a selection will delete everything
       from cursor position until the end of line.

I basically want want `s` does but only selecting and not deleting.

Do I have to use an Emacs API keybinding for this instead of a meow API keybinding?

**Comment by jgarte** [comment]
Ok, I figured out a workaround.

1. Make any type of selection first with meow.
2. Press `C-e` or `move-end-of-line` from `simple.el`.

Is this the recommended meow approach to doing that? I couldn't find another way yet.

https://chatgpt.com/share/003b63f3-3af9-4f02-9cdd-bf5ce63bab0f

**Comment by brabalan** [comment]
You can simply call `meow-end-of-thing` then choose `l` for line.

**Reply by jgarte** [reply]
Oh ok, that would be `]` followed by `l` on my qwerty keyboard. I think it is more ergonomic for me to do what I mentioned above but I'll see if I can rebind, get used to/reconsider. switch to dvorak or something else...

Thanks for the idea.

**Reply by jasonperry** [reply]
If you're selecting in order to copy, you can always kill to the end of line, then immediately yank. The text will still be in the kill ring to paste somewhere else.

**Reply by okamsn** [reply]
If you are copying to the end of the line, I wrote a small command for that as a fallback for `meow-save` when there is no current selection:

```emacs-lisp
  (defun my-meow-save-to-end-of-line (arg)
    "Copy the text from point to the end of the line.
If ARG, copy to start of line."
    (interactive "P")
    (save-excursion
      (let ((beg (point))
            (end (if arg
                     (if visual-line-mode
                         (save-mark-and-excursion
                           (beginning-of-visual-line 1)
                           (point))
                       (pos-bol))

                   (if visual-line-mode
                       (save-mark-and-excursion
                         (end-of-visual-line 1)
                         (point))
                     (pos-eol)))))
        (meow--select (meow--make-selection '(no-expand . char) beg end))
        (meow-save)
        (let ((pulse-delay 0.06))
          (pulse-momentary-highlight-region beg end 'region))
        (message "Yanked to %s of line." (if arg "start" "end")))))
```
